test/minikube/minikube_domain_resync_test.go

```diff
@@ -0,0 +1,193 @@
+			archiveVolumeClaims[claimName] = strings.TrimSpace(output)
+		} else {
+			assert.Equal(t, "TE", obj.Type, "archive-pvc label should only be absent for TEs")
+		}
+	}
+	return archiveVolumeClaims
+}
+
+type archive struct {
+	ArchiveId string `json:"archiveId"`
+	DbName    string `json:"dbName"`
+}
+
+func unmarshalArchives(s string) (err error, processes []archive) {
+	dec := json.NewDecoder(strings.NewReader(s))
+
+	for {
+		var obj archive
+		err = dec.Decode(&obj)
+		if err == io.EOF {
+			// all done
+			return nil, processes
+		}
+
+		if err != nil {
+			return
+		}
+
+		processes = append(processes, obj)
+	}
+}
+
+func checkArchives(t *testing.T, namespaceName string, adminPod string, numExpected int, numExpectedRemoved int) (archives []archive, removedArchives []archive) {
+	options := k8s.NewKubectlOptions("", "")
+	options.Namespace = namespaceName
+
+	// check archives
+	output, err := k8s.RunKubectlAndGetOutputE(t, options, "exec", adminPod, "--",
+		"nuocmd", "--show-json", "get", "archives", "--db-name", "demo")
+	assert.NilError(t, err, output)
+
+	err, archives = unmarshalArchives(output)
+	assert.NilError(t, err)
+	assert.Equal(t, numExpected, len(archives), output)
+
+	// check removed archives
+	output, err = k8s.RunKubectlAndGetOutputE(t, options, "exec", adminPod, "--",
+		"nuocmd", "--show-json", "get", "archives", "--db-name", "demo", "--removed")
+	assert.NilError(t, err, output)
+
+	err, removedArchives = unmarshalArchives(output)
+	assert.NilError(t, err)
+	assert.Equal(t, numExpectedRemoved, len(removedArchives), output)
+	return
+}
+
+func TestDomainResync(t *testing.T) {
+	testlib.AwaitTillerUp(t)
+	defer testlib.VerifyTeardown(t)
+	defer testlib.Teardown(testlib.TEARDOWN_ADMIN)
+
+	helmChartReleaseName, namespaceName := testlib.StartAdmin(t, &helm.Options{}, 1, "")
+
+	admin0 := fmt.Sprintf("%s-nuodb-cluster0-0", helmChartReleaseName)
+
+	defer testlib.Teardown(testlib.TEARDOWN_DATABASE) // ensure resources allocated in called functions are released when this function exits
+
+	testlib.StartDatabase(t, namespaceName, admin0, &helm.Options{
+		SetValues: map[string]string{
+			"database.sm.resources.requests.cpu":    "0.25",
+			"database.sm.resources.requests.memory": testlib.MINIMAL_VIABLE_ENGINE_MEMORY,
+			"database.te.resources.requests.cpu":    "0.25",
+			"database.te.resources.requests.memory": testlib.MINIMAL_VIABLE_ENGINE_MEMORY,
+		},
+	})
+
+	originalArchiveVolumeClaims := verifyProcessLabels(t, namespaceName, admin0)
+	assert.Equal(t, 1, len(originalArchiveVolumeClaims))
+	originalArchiveId := ""
+	for _, archiveId := range originalArchiveVolumeClaims {
+		originalArchiveId = archiveId
+	}
+	assert.Assert(t, originalArchiveId != "")
+
+	// update replica count
+	options := k8s.NewKubectlOptions("", "")
+	options.Namespace = namespaceName
+	output, err := k8s.RunKubectlAndGetOutputE(t, options, "get", "statefulset", "-o", "jsonpath={.items[*].metadata.name}")
+	assert.NilError(t, err, output)
```

Comment:
I am not a huge fan of these "output parsing" get functions. I prefer using the K8s API to list/grab the objects. There is the `StatefulSetList` class which might be a good start.

Reply:
I wasn't able to find a `GetStatefulSet()` function, but maybe it is because my Golang code navigation in Emacs was busted.